### ISSUE TITLE
Point jumps around screen when the pop-up appears

### ISSUE THREAD

**Issue by jaouadmourtada** [company-mode/company-mode#1336]
Hi,

I have been getting a strange bug with Company lately (it started about a few months ago, but has recently started to become unbearable).

The behavior can be seen on the caption attached HERE: ![bug](https://user-images.githubusercontent.com/15709599/183529331-1fbc7ca9-a2c0-4a9e-b81f-0762b4f3596f.gif); the output of the command `M-x company-diag` (in particular, my current setup) is given below.

Roughly speaking, in some circumstances, when the Company pop-up appears, the cursor suddenly moves around the buffer. This makes editing impossible and I have to turn off company when this happens.
I should mention that this occurs mainly when the current point is about at the middle of the buffer vertically.

I mainly use Emacs in Auctex-mode, and this is where the described issue occurs.

Thank you very much for your contribution to Company and for your help in this issue. I'd be happy to provide further details or perform some tests.

In the meantime, any "hot fix" would be really appreciated, such as how to disable the pop-up (but keep the one-line autocompletion).

Best wishes,
J

#### Output of the command `M-x company-diag`

Emacs 28.1 (aarch64-apple-darwin21.1.0) of 2022-05-11 on armbob.lan
Company 0.9.13

company-backends: ((company-auctex-macros company-auctex-symbols company-auctex-environments)
 company-auctex-bibs company-auctex-labels company-bbdb company-semantic company-cmake company-capf company-clang company-files
 (company-dabbrev-code company-gtags company-etags company-keywords)
 company-oddmuse company-dabbrev)

Used backend: company-dabbrev
Major mode: latex-mode
Prefix: ""

Completions:
[skipped for brevity; contains a list of 2000+ words]

**Comment by dgutov** [comment]
Hi! Is that the version installed from GNU ELPA?

If so, any change you could try the development version first? You can install it from MELPA.

**Comment by jaouadmourtada** [comment]
Hi!
The version I have is Company 20220731.2333 from MELPA.

**Comment by jaouadmourtada** [comment]
Hi! 
Just a quick update: after doing a few tests, this behavior does not seem to come from Company itself: it only appears when global-linum-mode is activated.
So I don't know to which extent this issue is within the scope of company itself, rather it is due to the interaction between the company pop-up and linum-mode.

**Comment by jaouadmourtada** [comment]
So since the issue in my case was due to interaction with linum-mode, and since auto-completion is (to me) much more useful than displaying line numbers, a simple fix was to simply de-activate linum-mode.

Besides, linum-mode leads to other performance issues (for instance, when used together with folding), and also seems to be somewhat deprecated: this page https://www.emacswiki.org/emacs/LineNumbers suggests that "display-line-numbers-mode" is the favored line numbering mode for newer versions of emacs.

All of this to say that there may not be any "issue" to be fixed. Perhaps one could add a warning somewhere that linum-mode interacts badly with company (leading to the pointer jumping around screen), and suggest either to ditch line numbering or to opt for display-line-numbers-mode.
--- 
For the record, my linum configuration was:
(global-linum-mode t)
(setq linum-format "%3d")

**Comment by dgutov** [comment]
`(setq display-line-numbers-width 3)` should correspond to the above value of `linum-format`.

**Comment by jaouadmourtada** [comment]
Indeed this fixes the issue. Thanks!

**Comment by dgutov** [comment]
Glad you got this working better. :+1: 

Now, I wouldn't mind actually to look into fixing this for linum, or at least to dig deeper first: maybe it's something that would affect other features/modes/etc, just in rare cases that are annoying to reproduce and consequently go unreported. If it's something tricky and linum-specific, then probably not.

Anyway, if you sometime manage to write down a step-by-step repro how to reproduce this, with specific file, and attach it here (the file and the steps), I'll be sure to take a look.

No hurry anyway. :-)

**Comment by jaouadmourtada** [comment]
Sure!

Here is a setup where it appears in my case:

With this init file (put in .txt as github does not accept .el)
[init.txt](https://github.com/company-mode/company-mode/files/9307960/init.txt)

and this sample text file:
[sample.txt](https://github.com/company-mode/company-mode/files/9307968/sample.txt)

Instructions:
open the "sample" text file, and go to the portion with "%%%". 
Then, center vertically (C-l).
Then, start typing a word starting with "con". 
At this point, the cursor should jump across screen.
With these steps, this is what I get:
![jump](https://user-images.githubusercontent.com/15709599/184115447-0c9ea129-4643-4b10-9ddd-f44c1a4bbbf3.gif)

===
Output of the command M-x company-diag:
Emacs 28.1 (aarch64-apple-darwin21.1.0) of 2022-05-11 on armbob.lan
Company 0.9.13

company-backends: (company-bbdb company-semantic company-cmake company-capf company-clang company-files
	      (company-dabbrev-code company-gtags company-etags company-keywords)
	      company-oddmuse company-dabbrev)

Used backend: company-dabbrev
Major mode: text-mode
Prefix: ""
Completions:
.....

**Comment by dgutov** [comment]
Thanks! For future reference: any unnecessary step in your reproduction scenario (like `aggressive-indent-mode` or `visual-line-mode`) means I end up spending extra time making sure they are not the culprit.

Anyway, it does seem specifc to `linum-mode`. I've tinkered with this for a whole but haven't managed to dig deep enough to figure out the exact problem. But it installs after-modification (and other) hooks which work by inserting overlays with `before-string` property. There are other conditions for the bug to occur, like the window to be short enough: apparently that forces its scrolling during redisplay, and that, in turn, moves point.

To compare to a similar package which also uses margins, I installed [nlinum](https://elpa.gnu.org/packages/nlinum.html), a later, apparently more efficient incarnation of this package. The difference is it performs this work during fontification instead. And it doesn't seem to trigger the problem.

So, after all, I simply recommend to use either of the alternatives: `display-line-numbers-mode` or `nlinum`. The former is said to be the fastest option, but I have no personal experience to confirm that claim.
